### PR TITLE
Feature/decorator

### PR DESCRIPTION
**Pull request by M-Anwar-Hussaini**
# In the Pull Request the followings are done:
- [x] Create `Nameable` class.
- [x] Create `Decorator` class.
- [x] Create `TrimmerDecorator` class.
- [x] Create `CapitalizeDecorator` class.
- [x] Implement the `correct_name` methode.

### CODE REVIEW

File: src/person.rb [hunk 1]
```diff
@@ -3,20 +3,24 @@ class Person
...
     @parent_permission = parent_permission
   end
 
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  def correct_name
+    @name
+  end
```
Comment (Ibizugbe) [inline src/person.rb:20]: Nice job implementing a correct name method to return the name attribute😉 But it is also a requirement that the person class inherits from the nameable😉.

image

File: src/trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+require_relative 'decorator'
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name[0, 10]
```
Comment (Ibizugbe) [inline src/trimmer_decorator.rb]: This method ought to check if the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word😉. this has not correctly been implemented here👍🏼. Kindly do well to implement it properly👍🏼.

```suggestion
@nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name

```

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @M-Anwar-Hussaini 👋🏼👋🏼👋🏼, 

require_changes


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] Check if the `Decorator` class inherits from Namable👍🏼.
- [x] The constructor a namable object from params is assigned to an instance variable👍🏼.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KanzaTahreem — APPROVED** [review]
# Approved 🙌

Hi @M-Anwar-Hussaini 

Good job implementing the suggested changes 💪 **_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**



## Highlights 🎯
There are some excellent points that you have made perfectly. Keep up the great work! 👏

-  ✅ Class is decorated
-  ✅ `Person` class inherits from `Nameanle`
-  ✅ Implemented reviewer's suggestions
-  ✅ Linters are working properly

## Optional suggestions ❕
N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**